### PR TITLE
style(typing): enable most disabled pyright checks

### PR DESCRIPTION
**Pull request by lengau**
It might be easier to review this commit by commit, which goes rule by rule

### CODE REVIEW

File: charmcraft/parts/reactive.py [hunk 1]
```diff
@@ -107,7 +108,7 @@ class ReactivePlugin(plugins.Plugin):
     properties_class = ReactivePluginProperties
     validator_class = ReactivePluginEnvironmentValidator
 
-    @override
+    @overrides.override
```
Comment (sergiusens) [inline charmcraft/parts/reactive.py:111]: why did you use typing override and then switch to this?

File: pyproject.toml [hunk 2]
```diff
@@ -150,6 +150,7 @@ ignore = [
 include = [
     "charmcraft",
     "tests/unit",
+    "tests/integration",
     # "tests",  # TODO: enable tests
```
Comment (sergiusens) [inline pyproject.toml:154]: is this TODO still valid, can we get rid of it now?

Comment (lengau) [reply pyproject.toml]: There are legacy tests outside of those structures, but they can probably be left. TODO removed.

File: charmcraft/parts/reactive.py (hunk 1 above)
Comment (lengau) [reply charmcraft/parts/reactive.py:111]: Working towards consistency with snapcraft/rockcraft, and this provides more/better functionality than the implementation in `typing`

**Pull request merged**